.ci/Jenkinsfile_cloud

```diff
@@ -19,51 +21,44 @@ pipeline {
+                    env.NUMBER_DEPLOYMENTS = Integer.valueOf(params.NUMBER_DEPLOYMENTS) > MAX_DEPLOYMENTS ? MAX_DEPLOYMENTS.toString() : params.NUMBER_DEPLOYMENTS
                 }
             }
         }
         stage ('Run tests on cloud') {
             steps {
-                echo "NUMBER_TEST_RUNS = ${env.NUMBER_TEST_RUNS}"
+                echo "All scenarios will executed ${env.NUMBER_TEST_RUNS} times on ${env.NUMBER_DEPLOYMENTS} deployments"
                 script {
-                    def testRuns = Integer.valueOf(env.NUMBER_TEST_RUNS)
-                    def loopCount = (testRuns/testsInParallel) > 1 ? testRuns/testsInParallel : 1
-                    def concurrentThreads = testRuns > testsInParallel ? testsInParallel : testRuns
-                    for (def j = 1; j <= loopCount; j++) {
-                        for (def i = 1; i <= concurrentThreads; i++) {
-                            threads[i] = {
-                                docker.image(mavenImage).withRun('-v ${HOME}/.m2:/root/.m2') {
-                                    for (def attempt = 1; attempt <= 3; attempt++) {
-                                        try {
-                                            withVaultSecret(secret: 'secret/kibana-issues/dev/cloud-staging-api-key', secret_field: 'value', variable_name: 'API_KEY') {
-                                                sh """./kibana-load-testing/scripts/deploy_and_test.sh -v '${params.STACK_VERSION}' -c '${params.DEPLOY_CONFIG}' -s '${params.SIMULATION}'"""
-                                            }
-                                            loadPipeline.uploadGatlingReport(i * j)
-                                            if (params.INGEST_RESULTS.toBoolean()) {
-                                                withVaultSecret(secret: 'secret/kibana-issues/prod/coverage/elasticsearch', secret_field: 'host', variable_name: 'HOST_FROM_VAULT') {
-                                                    withVaultSecret(secret: 'secret/kibana-issues/prod/coverage/elasticsearch', secret_field: 'username', variable_name: 'USER_FROM_VAULT') {
-                                                        withVaultSecret(secret: 'secret/kibana-issues/prod/coverage/elasticsearch', secret_field: 'password', variable_name: 'PASS_FROM_VAULT') {
-                                                            sh '''./kibana-load-testing/scripts/ingest_results.sh'''
-                                                        }
-                                                    }
-                                                }
-                                            }
-                                            break
-                                        } catch (Exception e) {
-                                            if (attempt == 3) {
-                                                throw new Exception("Could not succeed after ${attempt} attempts")
```

Comment:
My guess it was added to retry cloud deployment specifically. But it looks like when it fails, it fails even with the retry logic, so it is better to fail fast and communicate with Cloud team

Reply:
Yeah I agree this is probably unnecessary now with using a single deployment 👍